dateutil/parser/isoparser.py

```diff
@@ -0,0 +1,424 @@
+        as `+00:00`) will also be represented as :class:`dateutil.tz.tzutc`.
+
+        :param dt_str:
+            A string or stream containing only an ISO-8601 datetime string
+
+        :param common_only:
+            If true, parsing the uncommon formats will throw an error.
+
+        :return:
+            Returns a :class:`datetime.datetime` representing the string.
+            Unspecified components default to their lowest value, with the
+            exception of year, which will use the value passed to the
+            ``default_year`` parameter of the method's bound
+            :class:`Isoparser` instance. If that
+            would produce an invalid date (e.g. ``'--02-29'`` parsed with a
+            non-leap-year default date), the default will be the last leap
+            year to occur before the default year.
+        """
+        if common_only:
+            components, pos = self._parse_isodate_common(dt_str)
+        else:
+            components, pos = self._parse_isodate(dt_str)
+
+        if len(dt_str) > pos:
+            if dt_str[pos:pos + 1] == self._sep:
+                components += self._parse_isotime(dt_str[pos + 1:])
+            else:
+                raise ValueError('String contains unknown ISO components')
+
+        return datetime(*components)
+
+    @_takes_ascii
+    def parse_isodate(self, datestr):
+        """
+        Parse the date portion of an ISO string.
+
+        :param datestr:
+            The string portion of an ISO string, without a separator
+
+        :return:
+            Returns a :class:`datetime.date` object
+        """
+        components, pos = self._parse_isodate(datestr)
+        if pos < len(datestr):
+            raise ValueError('String contains unknown ISO ' +
+                             'components: {}'.format(datestr))
+        return date(*components)
+
+    @classmethod
+    @_takes_ascii
+    def parse_isotime(cls, timestr):
+        """
+        Parse the time portion of an ISO string.
+
+        :param timestr:
+            The time portion of an ISO string, without a separator
+
+        :return:
+            Returns a :class:`datetime.time` object
+        """
+        return time(*cls._parse_isotime(timestr))
+
+    @classmethod
+    @_takes_ascii
+    def parse_tzstr(cls, tzstr, zero_as_utc=True):
+        """
+        Parse a valid ISO time zone string.
+
+        See :func:`Isoparser.isoparse` for details on supported formats.
+
+        :param tzstr:
+            A string representing an ISO time zone offset
+
+        :param zero_as_utc:
+            Whether to return :class:`dateutil.tz.tzutc` for zero-offset zones
+
+        :return:
+            Returns :class:`dateutil.tz.tzoffset` for offsets and
+            :class:`dateutil.tz.tzutc` for ``Z`` and (if ``zero_as_utc`` is
+            specified) offsets equivalent to UTC.
+        """
+        return cls._parse_tzstr(tzstr, zero_as_utc=zero_as_utc)
```

Comment:
This strikes me as redundant.  What's the benefit of having both of these?

Reply:
`parse_tzstr` is a public function, `_parse_tzstr` is the private implementation.

In this case, it's useful because `parse_tzstr` has the `@_takes_ascii` decorator and `_parse_tzstr` doesn't (because it is only called from other functions that already have the `@_takes_ascii` decorator, and as such it prevents an unnecessary call).

Reply:
The `_takes_ascii` point is compelling; the private vs public doesn't seem like a meaningful distinction in this case.

Reply:
Private means it's an implementation detail. You can't count on it existing, but the current implementation requires it. If we later find a different way to do this, the only thing we're guaranteeing is that the public interface has a `parse_tzstr` function that takes something ascii-like.